app/models/git_repository.rb

```diff
@@ -114,8 +107,22 @@ def file_changed?(sha1, sha2, file)
     executor.execute!("cd #{pwd}", "git diff --quiet --name-only #{sha1}..#{sha2} #{file}")
   end
 
+  def file_content(sha, file)
+    raise ArgumentError, "Need a sha, but #{sha} (#{sha.size}) given" unless sha.size == 40
```

Comment:
nice, taken care of in next PR

On Mon, May 2, 2016 at 9:17 AM, Jon Moter <notifications@github.com> wrote:

> In app/models/git_repository.rb
> <https://github.com/zendesk/samson/pull/918#discussion_r61761260>:
>
> > @@ -114,8 +107,22 @@ def file_changed?(sha1, sha2, file)
> >      executor.execute!("cd #{pwd}", "git diff --quiet --name-only #{sha1}..#{sha2} #{file}")
> >    end
> >
> > +  def file_content(sha, file)
> > +    raise ArgumentError, "Need a sha, but #{sha} (#{sha.size}) given" unless sha.size == 40
>
> FYI, the Build model has a validation that the git_sha attribute is a
> valid SHA1. Might want to extract out the logic for checking that something
> is a valid SHA and use that, rather than checking length here, and using a
> regex there.
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly or view it on GitHub
> <https://github.com/zendesk/samson/pull/918/files/a1302c1aa7a6e1c2937dc80b57a57eec9b3309cc#r61761260>
>